shared/extension/Behaviour.hpp

```diff
@@ -243,81 +249,124 @@ namespace extension::behaviour {
-         * @param data       the data element of the task
-         * @param name       a string identifier to help with debugging
-         * @param priority   the priority that this task is to be executed with
-         * @param optional   if this task is an optional task and can be taken over by lower priority non optional tasks
+         * @tparam DSL the DSL from NUClear
+         *
+         * @param reaction the reaction that is having this needs condition bound to it
          */
-        static void emit(NUClear::PowerPlant& powerplant,
-                         std::shared_ptr<T> data,
-                         const std::string& name = "",
-                         const int& priority     = 1,
-                         const bool& optional    = false) {
-
-            // Work out who is sending the task so we can determine if it's a subtask
-            const auto* task     = NUClear::threading::ReactionTask::get_current_task();
-            uint64_t reaction_id = (task != nullptr) ? task->parent.id : -1;
-            uint64_t task_id     = (task != nullptr) ? task->id : -1;
-
-            NUClear::dsl::word::emit::Direct<T>::emit(powerplant,
-                                                      std::make_shared<commands::DirectorTask>(typeid(T),
-                                                                                               reaction_id,
-                                                                                               task_id,
-                                                                                               data,
-                                                                                               name,
-                                                                                               priority,
-                                                                                               optional));
+        template <typename DSL>
+        static inline void bind(const std::shared_ptr<NUClear::threading::Reaction>& reaction) {
+            reaction->reactor.emit<NUClear::dsl::word::emit::Direct>(
+                std::make_unique<commands::NeedsExpression>(reaction->id, typeid(Provider)));
         }
     };
 
     /**
-     * This is a special task that should be emitted when a Provider finishes the task it was given.
-     * When this is emitted the director will re-execute the Provider which caused this task to run.
+     * Adds a Uses object to the callback to allow access to information from the context of this provider.
+     *
+     * The uses object provides information about what would happen if you were to emit a specific task from this
+     * provider as well as what has happened in previous calls.
+     *
+     * It has two main uses
+     *   It allows you to see what would happen if you emitted a task (run, block proxy, etc)
+     *   It allows you to see what has happened in previous calls, e.g. are we re-running because that was emitted done
```

Comment:
```suggestion
     *   It allows you to see what has happened in previous calls, e.g. are we re-running because what was emitted is done
```
I'm not sure if this is right but the sentence didn't make sense.

Reply:
You're right it didn't!